vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,712 @@
+
+### The modified `app/view/datatable.R` module
+
+As the `toggle` module returns the filtered data, it is used in the `datatable` module so that the data is displayed on a table accordingly. Hence, our `datatable` module is modified as shown below -
+
+```
+# app/view/datatable.R
+
+box::use(
+  shiny[moduleServer, NS, reactive, renderUI, span, uiOutput],
+  shiny.fluent[DetailsList],
+  tibble[tibble],
+)
+
+columns <- tibble(
+  fieldName = c("rep_name", "date", "deal_amount", "city", "is_closed"),
+  name = c("Sales Rep", "Date", "Deal Amount", "City", "Is Closed?")
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  uiOutput(ns("datatable"))
+}
+
+#' @export
+server <- function(id, data) {
+  moduleServer(id, function(input, output, session) {
+    output$datatable <- renderUI({
+      DetailsList(items = data(), columns = columns)
+    })
+  })
+}
+```
+Please notice, that inside the server function, the response obtained from the `toggle` module earlier is also passed on to be used in displaying the data in a table. 
+
+We are not returning any information from this module to the main module as it's sole purpose is to display given data in a table with desired columns in an easily readable form.
+
+### The `app/view/plot.R` module
+
+This module also obtains the data returned from the `toggle` module and renders a barchart using the `app/logic/barchart.R` logic.
+
+```
+# app/view/plot.R
+
+box::use(
+  plotly[plotlyOutput, renderPlotly],
+  shiny[moduleServer, NS],
+)
+
+box::use(
+  app/logic/barchart[generate_barchart],
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  plotlyOutput(ns("barchart"))
+}
+
+#' @export
+server <- function(id, data) {
+  moduleServer(id, function(input, output, session) {
+    output$barchart <- renderPlotly({
+      generate_barchart(data())
+    })
+  })
+}
+```
+### The modified `app/main.R` module
+
+So far we have created the modules that we need for the app's contents. 
+
+To make them work, we would need to call them in the main module. Here is how the main module has changed -
+
+```
+# app/view/main.R
+
+box::use(
+  shiny[moduleServer, NS, reactive, span],
+  shiny.fluent[fluentPage, Stack, Text],
+)
+
+box::use(
+  app/view/datatable,
+  app/view/toggle,
+  app/view/plot,
+)
+
+#' @export
+ui <- function(id) {
+  ns <- NS(id)
+  fluentPage(      
+    Text("Sales Data Dashboard", variant = "xxLarge"),
+    toggle$ui(ns("toggle")),
+    Stack(
+      horizontal = TRUE,
+      tokens = list(padding = 20, childrenGap = 10),
+      span(class = "ms-depth-8", datatable$ui(ns("datatable"))),
+      span(class = "ms-depth-8", plot$ui(ns("plot")))
+    )
+  )
+}
+
+#' @export
+server <- function(id) {
+  moduleServer(id, function(input, output, session) {
+    data <- reactive({
+      toggle$server("toggle")      
+    })
```

Comment:
Wrapping a module call in a `reactive` is redundant. The module itself should return a reactive value.